admittance_controller/include/admittance_controller/admittance_rule_impl.hpp

```diff
@@ -0,0 +1,534 @@
+        kinematics_ = std::unique_ptr<kinematics_interface::KinematicsBaseClass>(
+            kinematics_loader_->createUnmanagedInstance(parameters_->kinematics_.plugin_name_));
+        if (!kinematics_->initialize(node, parameters_->kinematics_.tip_)) {
+          return controller_interface::return_type::ERROR;
+        }
+      }
+      catch (pluginlib::PluginlibException &ex) {
+        RCLCPP_ERROR(rclcpp::get_logger("AdmittanceRule"), "Exception while loading the IK plugin '%s': '%s'",
+                     parameters_->kinematics_.plugin_name_.c_str(), ex.what());
+        return controller_interface::return_type::ERROR;
+      }
+    } else {
+      RCLCPP_ERROR(rclcpp::get_logger("AdmittanceRule"),
+                   "A differential IK plugin name was not specified in the config file.");
+      return controller_interface::return_type::ERROR;
+    }
+
+    return controller_interface::return_type::OK;
+  }
+
+  controller_interface::return_type AdmittanceRule::reset() {
+    // reset all values back to zero
+
+    //allocate dynamic buffers
+    joint_buffer_vec_.assign(num_joints_, 0.0);
+    transform_buffer_vec_.assign(16, 0.0);
+    cart_buffer_vec_.assign(6, 0.0);
+    jacobian_buffer_vec_.assign(6 * num_joints_, 0.0);
+
+    state_message_.error_joint_state.positions.resize(num_joints_, 0.0);
+    state_message_.error_joint_state.velocities.resize(num_joints_, 0.0);
+    state_message_.admittance_rule_calculated_values.positions.resize(6, 0.0);
+    state_message_.admittance_rule_calculated_values.velocities.resize(6, 0.0);
+    state_message_.admittance_rule_calculated_values.accelerations.resize(6, 0.0);
+
+
+    // admittance state vectors in joint space
+    joint_pos_ = Eigen::Matrix<double, Eigen::Dynamic, 1, Eigen::ColMajor>(num_joints_);
+    joint_vel_ = Eigen::Matrix<double, Eigen::Dynamic, 1, Eigen::ColMajor>(num_joints_);
+    joint_acc_ = Eigen::Matrix<double, Eigen::Dynamic, 1, Eigen::ColMajor>(num_joints_);
+
+    // transforms
+    ee_transform_.setIdentity();
+    reference_ft_transform_.setIdentity();
+    sensor_transform_.setIdentity();
+    control_transform_.setIdentity();
+    cog_transform_.setIdentity();
+    world_transform_.setIdentity();
+    ft_transform_.setIdentity();
+
+    // admittance values
+    admittance_position_.setIdentity();
+    admittance_velocity_.setZero();
+    admittance_acceleration_.setZero();
+    wrench_world_.setZero();
+    jacobian_ = Eigen::Matrix<double, 6, Eigen::Dynamic, Eigen::ColMajor>(6, num_joints_);
+    jacobian_.setZero();
+    identity = Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::ColMajor>(num_joints_, num_joints_);
+    identity.setIdentity();
+    damping_.setZero();
+    mass_.setOnes();
+    mass_inv_.setZero();
+    stiffness_.setZero();
+    selected_axes_.setZero();
+
+    // parameters
+    if (parameter_handler_->params_.enable_parameter_update_without_reactivation_) {
```

Comment:
There are two cases. If dynamic updating is disabled, then we need to make a copy of the parameters. Otherwise, we can point to the parameters in the `parameter_handler_`.

Reply:
I still don't understand this. We should clarify what dynamic updating means. For me, this means that you can use dynamic reconfigure to update parameters. But those should be first updated in the controller when the controller is re-activated. I don't think we want to support changing or parameters when controller is in active state.